Geometry/HGCalCommonData/src/HGCalCellOffset.cc

```diff
@@ -0,0 +1,138 @@
+        double x3 = 0.5 * cellY_[k];
+        double y3 = (cellY_[k] * sqrt3By2_) - (2 * mouseBiteCut_ / 3) - (guardRingOffset_ / (3 * sqrt3By2_));
+        cellArea[k][j] = totalArea - cutArea1 - cutArea2 - cutArea3;
+        double xMag = ((35.0 * cellY_[k] / 132.0) * totalArea - (cutArea1 * x1) - (cutArea2 * x2) - (cutArea3 * x3)) /
+                      (cellArea[k][j]);
+        double yMag =
+            ((5.0 * cellY_[k] / (44.0 * sqrt3_)) * totalArea - (cutArea1 * y1) - (cutArea2 * y2) - (cutArea3 * y3)) /
+            (cellArea[k][j]);
+        tempOffsetX.emplace_back(std::vector<double>({xMag,
+                                                      -1.0 * (0.5 * xMag - sqrt3By2_ * yMag),
+                                                      (-0.5 * xMag + sqrt3By2_ * yMag),
+                                                      xMag,
+                                                      (-0.5 * xMag - sqrt3By2_ * yMag),
+                                                      -1.0 * (0.5 * xMag + sqrt3By2_ * yMag)}));
+        tempOffsetY.emplace_back(std::vector<double>({yMag,
+                                                      (0.5 * yMag + sqrt3By2_ * xMag),
+                                                      (-0.5 * yMag - sqrt3By2_ * xMag),
+                                                      -yMag,
+                                                      (-0.5 * yMag + sqrt3By2_ * xMag),
+                                                      (0.5 * yMag - sqrt3By2_ * xMag)}));
+      } else if (j == 1) {
+        double totalArea = (5.0 * sqrt3_ / 4.0) * std::pow(cellY_[k], 2);
+        double cutArea = cellY_[k] * sqrt3_ * guardRingOffset_;
+        cellArea[k][j] = totalArea - cutArea;
+        double offMag = (((-2.0 / 15.0) * totalArea * cellY_[k]) - ((cellY_[k] - (0.5 * guardRingOffset_)) * cutArea)) /
+                        (cellArea[k][j]);
+        tempOffsetX.emplace_back(
+            std::vector<double>({-0.5 * offMag, -offMag, -0.5 * offMag, 0.5 * offMag, offMag, 0.5 * offMag}));
+        tempOffsetY.emplace_back(std::vector<double>(
+            {-sqrt3By2_ * offMag, 0.0, sqrt3By2_ * offMag, sqrt3By2_ * offMag, 0.0, -sqrt3By2_ * offMag}));
+      } else if (j == 2) {
+        double totalArea = (7.0 * sqrt3_ / 4.0) * std::pow(cellY_[k], 2);
+        double cutArea = cellY_[k] * sqrt3_ * guardRingOffset_;
+        cellArea[k][j] = totalArea - cutArea;
+        double offMag =
+            (((5.0 / 42.0) * totalArea * cellY_[k]) - ((cellY_[k] - (0.5 * guardRingOffset_))) * (cutArea)) /
+            (cellArea[k][j]);
+        tempOffsetX.emplace_back(
+            std::vector<double>({-0.5 * offMag, -offMag, -0.5 * offMag, 0.5 * offMag, offMag, 0.5 * offMag}));
+        tempOffsetY.emplace_back(std::vector<double>(
+            {-sqrt3By2_ * offMag, 0.0, sqrt3By2_ * offMag, sqrt3By2_ * offMag, 0.0, -sqrt3By2_ * offMag}));
+      }
+    }
```

Comment:
the whole code is flooded with ad-hoc, hard-coded numbers that have no immediate meaning to the inexperienced reader. Could you make the code and the constants "named" somehow to help understand the code?

Reply:
They represent coordinate transformation, the formula of which is added in comments